File: 01-FUNDAMENTOS_DE_PROGRAMACAO/TypeScript/Exercicio12.ts
/* 
12 - Faça um programa que contenha uma função que recebe dois números
positivos e retorna a soma dos N números inteiros existentes entre eles (inclusive).
Use a função no programa principal 
*/

function somaXY(x: number, y: number) {
    let soma = 0
    if (y > x) {
        for (let i = x; i <= y; i++)
            soma = soma + i
    }
    else {
        for (let i = y; i <= x; i++)
            soma = soma + i
    }
    return soma
}

let x = 2; let y = 5;
const funcao = somaXY(x, y)
console.log('\n O valor da soma eh: ', funcao)

File: 01-FUNDAMENTOS_DE_PROGRAMACAO/C/Exercicio 12.c
/* 
12 - Faça um programa que contenha uma função que recebe dois números
positivos e retorna a soma dos N números inteiros existentes entre eles (inclusive).
Use a função no programa principal 
*/

#include<stdio.h>

int soma (int x, int y)
{
    int soma=0, i;
    if (y>x){
    for (i=x; i<=y; i++){
        soma=soma+i;
    }
    return (soma);
    }
    else {
        for (i=y; i<=x; i++){
        soma=soma+i;
    }
    return (soma);
    }
}

int main ()
{
    int x, y, funcao;
    printf ("Entre com dois numeros: \n");
    scanf ("%d %d", &x, &y);
    funcao = soma (x, y);
    printf ("\n O valor da soma eh: %d\n",funcao);
}
